components/OverlayComponent.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <no-ssr>
+    <div class="overlay-component">
+
+      <div v-if="shouldShowWeb3Dialog">
+        <div v-if="checkIsMobileClient()">
+          <!-- Trust Dialog start (show when mobile) -->
+          <v-dialog
+            v-model="isDialogOpen"
+            max-width="290"
+            @keydown.esc="isDialogOpen = false"
+          >
+            <v-card>
+              <v-card-title class="headline">
+                Wallet App required
+              </v-card-title>
+            </v-card>
+            <v-btn dark flat @click.native="isDialogOpen = false">Close</v-btn>
+          </v-dialog>
+          <!-- Trust Dialog end -->
+        </div>
+        <div v-else>
+          <!-- Chrome Dialog start (ask when user use browser like Safari) -->
+          <v-dialog
+            v-model="isDialogOpen"
+            max-width="290"
+            v-if="shouldShowChromeDialog"
+            @keydown.esc="isDialogOpen = false"
+          >
+            <v-card>
+              <v-card-title class="headline">
+                Chrome required
+              </v-card-title>
+            </v-card>
+            <v-btn dark flat @click.native="isDialogOpen = false">Close</v-btn>
+          </v-dialog>
+          <!-- Chrome Dialog end -->
+
+          <!-- Metamask related dialog start -->
+          <v-dialog
+            v-model="isDialogOpen"
+            max-width="290"
+            v-else-if="!!getWeb3Message"
+            @keydown.esc="isDialogOpen = false"
+          >
+            <v-card>
+              <v-card-title class="headline">
+                {{ metamaskDialogMessage }}
+              </v-card-title>
+            </v-card>
+            <v-btn dark flat @click.native="isDialogOpen = false">Close</v-btn>
+          </v-dialog>
+          <!-- Metamask related dialog end -->
+        </div>
+      </div>
+
+    </div>
+  </no-ssr>
+</template>
+<script>
+import { mapGetters } from 'vuex';
+
+import {
+  checkIsMobileClient,
+  checkIsDesktopChrome,
+} from '@/util/client';
+
+export default {
+  name: 'overlay-component',
```

Comment:
Please rename the `OverlayComponent` to `TheOverlayComponents`, as well as the class name